src/controllers/restatedeployment/reconcilers/knative.rs

```diff
@@ -0,0 +1,938 @@
+                        RESTATE_TAG_ANNOTATION: tag,
+                        RESTATE_REGISTERED_AT_ANNOTATION: chrono::Utc::now().to_rfc3339(),
+                    }
+                }
+            })),
+        )
+        .await?;
+
+    Ok(())
+}
+
+/// Update RestateDeployment status in-place
+/// This populates the status fields without patching to avoid race conditions
+/// The controller will apply the complete status in a single patch
+fn update_status(
+    status: &mut crate::resources::restatedeployments::RestateDeploymentStatus,
+    current_tag: &str,
+    config: &Configuration,
+    route: &Route,
+    revision: &Revision,
+    deployment_id: &str,
+) {
+    let url = route.status.as_ref().and_then(|s| s.url.clone());
+
+    let knative_status = KnativeDeploymentStatus {
+        current_tag: Some(current_tag.to_string()),
+        configuration_name: Some(config.name_any()),
+        route_name: Some(route.name_any()),
+        url,
+        latest_revision: Some(revision.name_any()),
+        deployment_id: Some(deployment_id.to_string()),
+    };
+
+    // Extract replica counts from Revision status
+    // Note: When scaled to zero, Knative may omit these fields
+    let (desired, actual, ready_replicas, available_replicas, unavailable_replicas) =
+        if let Some(rev_status) = &revision.status {
+            let desired = rev_status.desired_replicas.unwrap_or(0);
+            let actual = rev_status.actual_replicas.unwrap_or(0);
+            (
+                desired,
+                actual,
+                Some(actual),
+                Some(actual),
+                Some((desired - actual).max(0)),
+            )
+        } else {
+            // Revision has no status yet
+            (0, 0, Some(0), Some(0), Some(0))
+        };
+
+    // Update status fields in-place
+    status.knative = Some(knative_status);
+    status.replicas = actual;
+    status.desired_replicas = Some(desired);
+    status.ready_replicas = ready_replicas;
+    status.available_replicas = available_replicas;
+    status.unavailable_replicas = unavailable_replicas;
+    // Knative mode doesn't use label selectors (Knative manages pods directly)
+    status.label_selector = None;
+}
+
+const RESTATE_REMOVE_VERSION_AT_ANNOTATION: &str = "restate.dev/remove-version-at";
+
+/// Delete Configurations that are no longer needed
+/// Mirrors the pattern from cleanup_old_replicasets() in replicaset.rs:156-399
+#[allow(clippy::too_many_arguments)]
+pub async fn cleanup_old_configurations(
+    namespace: &str,
+    ctx: &Context,
+    rsd_uid: &str,
+    rsd: &RestateDeployment,
+    deployments: &std::collections::HashMap<String, bool>,
+    current_tag: &str,
+) -> Result<(i32, Option<chrono::DateTime<chrono::Utc>>)> {
+    // List all Configurations in the namespace
+    let config_api: Api<Configuration> = Api::namespaced(ctx.client.clone(), namespace);
+    let all_configs = config_api.list(&Default::default()).await?;
```

Comment:
in replicaset land we use a reflector cache for this query, to avoid a list on every reconcile. can we do the same here?